lib/phoenix/pubsub/adapter.ex

```diff
@@ -0,0 +1,71 @@
+defmodule Phoenix.PubSub.Adapter do
+  use Behaviour
+
+  @moduledoc """
+  Defines the behaviour of a PubSub adapter.
+
+  See `Phoenix.PubSub.PG2Adapter` for an example implementation.
+  """
+
+  @doc """
+  Starts the adapter
+  """
+  defcallback start_link(options :: list) :: {:ok, pid} |
+                                             :ignore |
+                                             {:error, reason :: term}
+
+  @doc """
+  Stops the adapter
+  """
+  defcallback stop() :: :ok | {:error, reason :: term}
+
+  @doc """
+  Creates a PubSub group for given topic to hold subscriptions
+  """
+  defcallback create(topic :: String.t) :: :ok | {:error, reason :: term}
+
+  @doc """
+  Checks if PubSub group exists for given topic
+  """
+  defcallback exists?(topic :: String.t) :: true | false | {:error, reason :: term}
```

Comment:
This one can be tricky too. My recommendation would be to slim down this behaviour module as much as possible. :D